controllers/authController.js

```diff
@@ -0,0 +1,58 @@
+const User = require('../models/User');
+const CustomError = require('../errors');
+//const { attachCookiesToResponse, createTokenUser } = require('../utils');
+
+const register = async (req, res) => {
+	const { email, name, password } = req.body;
+
+	const emailAlreadyExists = await User.findOne({ email });
+	if (emailAlreadyExists) {
+		throw new CustomError.BadRequestError('Email already exists');
+	}
+
+	const isFirstAccount = (await User.countDocuments({})) === 0;
+	const role = isFirstAccount ? 'admin' : 'user';
+
+	const user = await User.create({ name, email, password, role });
+	const tokenUser = createTokenUser(user);
+	attachCookiesToResponse({ res, user: tokenUser });
+	res.status(200).json({ user: tokenUser });
+};
+const login = async (req, res) => {
+	const { email, password } = req.body;
+
+	if (!email || !password) {
+		throw new CustomError.BadRequestError('Please provide email and password');
+	}
+	const user = await User.findOne({ email });
+
+	if (!user) {
+		throw new CustomError.BadRequestError('Invalid Credentials');
+	}
+	const isPasswordCorrect = await user.comparePassword(password);
+	if (!isPasswordCorrect) {
+		throw new CustomError.BadRequestError('Invalid Credentials');
+	}
+	//const tokenUser = createTokenUser(user);
+	//attachCookiesToResponse({ res, user: tokenUser });
+
+	res.status(200).json({ user: 'blah-blah' });
```

Comment:
This ain't right. But you decide whether to leave it for a future ticket or to actually add some sense to this. Or to just wrap everything here in a comment and to call it a day.